### ISSUE TITLE
Really difficult to hook into library

### ISSUE THREAD

**Issue by ctian1** [jossmac/react-toast-notifications#57]
for example:
- changing the default ToastContainer z-index requires a custom ToastContainer class to add a style prop
- making autoDismiss default requires a bunch of manipulation to override methods

**Comment by Vynlar** [comment]
I agree that having the ability to pass a default for `autoDismiss` to the provider itself would be very useful. My current workaround is to wrap the `useToasts` hook:
```
export const useToasts = () => {
  const { addToast, ...rest } = useLibraryToasts(); // the `useToasts` hook from the library

  return {
    ...rest,
    addToast: (content, options) =>
      addToast(content, {
        autoDismiss: true,
        ...options,
      }),
  };
};
```

**Comment by jossmac** [comment]
@ctian1 @Vynlar It seems to be unanimous that `autoDismiss` should be inherited. I'll try to get onto that this weekend.

Regarding `z-index` on the container; I think the [current solution](https://github.com/jossmac/react-toast-notifications/issues/45#issuecomment-515818086) is reasonable for now. I would love to implement a styles API, similar to what Jed and I have done with [react-select](https://react-select.com/styles), but it takes time and I have to prioritise my family.

Since there is a bunch of prior art (esp. with emotion) I would like to take this opportunity to mention that pull requests are welcome.

**Comment by jossmac** [comment]
Resolved by #75 

* * *

Available in `2.3.0`

* toasts will inherit `autoDismiss` from the provider
* add default z-index (`1000`) to the container
* add class names to default components
